src/methods/SANA.cpp

```diff
@@ -1428,10 +1428,14 @@ bool SANA::scoreComparison(double newAligEdges, double newInducedEdges, double n
 
 int SANA::aligEdgesIncChangeOp(uint source, uint oldTarget, uint newTarget) {
     int res = 0;
-    const uint n = G1AdjLists[source].size();
-    uint neighbor;
-    for (uint i = 0; i < n; ++i) {
-        neighbor = G1AdjLists[source][i];
+
+    vector<uint> v = G1AdjLists[source];
+    if(G1->hasSelfLoop(source)) {
```

Comment:
Again even calling this function may add uncessary expense. If SPARSE is false, just check the adjacencry matrix inline right here. Perform the function call only of SPARSE is true and it's necessary to perform the O(degree) search.